packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -92,15 +169,126 @@ class FlutterGoldenFileComparator implements GoldenFileComparator {
     }
     return true;
   }
+}
+
+/// A [FlutterGoldenFileComparator] for testing golden images with Skia Gold.
+///
+/// For testing across all platforms, the [SkiaGoldClient] is used to upload
+/// images for framework-related golden tests and process results. Currently
+/// these tests are designed to be run post-submit on Cirrus CI, informed by the
+/// environment.
+///
+/// See also:
+///
+///  * [GoldenFileComparator], the abstract class that
+///    [FlutterGoldenFileComparator] implements.
+///  * [FlutterGoldensRepositoryFileComparator], another
+///    [FlutterGoldenFileComparator] that tests golden images using the
+///    flutter/goldens repository.
+class FlutterSkiaGoldFileComparator extends FlutterGoldenFileComparator {
+  FlutterSkiaGoldFileComparator(
+    Uri basedir, {
+    FileSystem fs = const LocalFileSystem(),
+    Platform platform = const LocalPlatform(),
+  }) : super(
+    basedir,
+    fs: fs,
+    platform: platform,
+  ) {
+    _skiaClient = SkiaGoldClient(fs.directory(basedir));
```

Comment:
This could be set in the initializer as well and made `final`

Reply:
I don't know that I can do that since the SkiaGoldClient is not a const constructor. cc/ @goderbauer for offline chat.